Extract NTSC assets for gc-us

cadmic opened the pull request:
except text, which @Thar0 has been working on.

I created new XMLs for:
* gameplay_keep, based on https://discord.com/channels/688807550715560050/793267277859520563/880509593917194250. Many other unchanged XML files depend on gameplay_keep as an `<ExternalFile>`, so I moved this `<ExternalFile>` to Config.xml instead so it can change between versions. I think this probably slows down the build a bit but it's not too bad.
* Boss objects: title cards are smaller (2 languages instead of 3) which shifts everything after
* object_du and object_xc (different animations)
* spot20 (LLR credits cutscene?)
* Language-specific icons (do_action_static, icon_item_gameover_static, icon_item_static, item_name_static, map_name_static, place_title_cards, title_static)
* The NTSC-only `kanji` ROM segment (based on https://github.com/zeldaret/mm/pull/1536)

Dragorn421 commented on tools/ZAPDConfigs/gc-eu/Config.xml:
```diff
@@ -6,4 +6,5 @@
     <SpecialEntranceList File="../SpecialEntranceList.txt"/>
     <EnumData File="../EnumData.xml"/>
     <ExternalXMLFolder Path="assets/xml/"/>
+    <ExternalFile XmlPath="objects/gameplay_keep_pal.xml" OutPath="assets/objects/gameplay_keep/"/>
```
I'd rather see this in each xml that needs it, since not all assets reference gkeep, nor does it make sense for all assets to have gkeep available in their relevant memory map

Also new-assets co-extracts assets that reference each other this way in order to use the extra information for extraction. Making all assets co-extract with gkeep would prevent multiprocessing

cadmic replied:
Yeah this is not great, but otherwise I think we'd have to duplicate hundreds of XMLs to either reference `objects/gameplay_keep_pal.xml` or `objects/gameplay_keep.xml` depending on the version...

Dragorn421 replied:
Ah I didn't catch that
Yeah ok let's go like this (gkeep in global config) for now then. :(

Dragorn421 reviewed (commented):
Good job on the steady progress, thanks :)

Can't say I'm enthusiastic about the amount of duplication. Is this specific to introducing ntsc versions or can we expect even more?

cadmic commented:
AFAIK the NTSC/PAL assets are the same for all NTSC/PAL versions so this is basically the last round of duplication. The remaining differences are:

nintendo_rogo_static: changed after NTSC 1.0
object_mag: changed between N64/GC versions (also gc-jp and gc-jp-mq have different versions from the rest of the GC versions)

cadmic commented:
gameplay_keep diffs for easier review: https://discord.com/channels/688807550715560050/688851317593997489/1254952193128403127

Dragorn421 reviewed (approved):
Okay for now, again I didn't look closely at the xmls, we'll catch things to fix eventually and I'm sure it's fine :tm: .

FTR, two things I want to change later (with the new assets system):

1) reference gkeep per-asset instead of globally (this PR references it globally with `<ExternalFile XmlPath="objects/gameplay_keep_pal.xml" OutPath="assets/objects/gameplay_keep/"/>` in `tools/ZAPDConfigs/VERSION/Config.xml`).
The reason this PR references it globally is because it introduces two gkeep versions, one ntsc and one pal. If each xml had to reference it we currently would have to duplicate these xmls for ntsc and pal too.
In the future we'll probably add a better way to indicate referencing gkeep with some indirection rather than pointing to the xml directly

2) un-duplicate the xmls, this PR duplicates a lot of them because of changing offsets mostly, and some resources are also added/removed.
In the future we'll figure out a way to have finer grained version handling inside the xmls rather than completely duplicate them
(my personal preference for now would leverage xml namespaces as someone suggested on discord and could look like `<Resource ... Offset="0x120" gc_eu_mq:Offset="0x150" />` but this is up for discussion)

fig02 reviewed (approved):
also did not review too thoroughly
